client/src/components/game/carrier/CarrierDetail.vue

```diff
@@ -261,6 +302,36 @@ export default {
     viewOnMap (e) {
       GameContainer.map.panToCarrier(this.carrier)
     },
+    getEffectiveSpeed () {
```

Comment:
This method and the methods below could have their implementation in the `GameHelper`, that might be a better place for this game logic to sit in. Not fussed though.